pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -756,9 +749,8 @@ func (pl *DynamicResources) filterExtendedResources(state *stateData, pod *v1.Po
 		if rQuant == 0 {
 			continue
 		}
-
 		allocatable, okScalar := nodeInfo.GetAllocatable().GetScalarResources()[rName]
-		_, okDynamic := state.draExtendedResource.resourceToDeviceClass[rName]
+		okDynamic := cache.GetDeviceClass(rName) != ""
```

Comment:
Nit: `okDynamic` is not very descriptive. Perhaps `isBackedByDRA`?

Can also be done in a future refactoring PR (cc @alaypatel07).